api/v1alpha1/webhook.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// Maximum length of bundle name
+const maxNameLength = 40
+
+// log is for logging in this package.
+var bundlelog = logf.Log.WithName("bundle-resource")
+
+func (r *Bundle) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	bundlelog.Info("SetupWebhookWithManager")
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-core-rukpak-io-v1alpha1-bundle,mutating=false,failurePolicy=fail,sideEffects=None,groups=core.rukpak.io,resources=bundles,verbs=create;update,versions=v1alpha1,name=core.rukpak.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &Bundle{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *Bundle) ValidateCreate() error {
+	bundlelog.V(1).Info("validate create", "name", r.Name)
```

Comment:
Should we bump the log verbosity of this to a higher level? Is there a way we can inject a logger such that we don't rely on a global logger for this package? Can we wrap this ValidateCreate method and inject a logger as a parameter?

Reply:
It is `DEBUG` level now.
`1.648259706745181e+09   DEBUG   bundle-resource validate create {"name": "combo-78901234567890123456789012345678901"}`
Another log entry is generated by the controller-runtime.  It's also `DEBUG` level.
`1.6482597067455616e+09  DEBUG   controller-runtime.webhook.webhooks     wrote response  {"webhook": "/validate-core-rukpak-io-v1alpha1-bundle", "code": 403, "reason": "bundle name combo-78901234567890123456789012345678901 is too long: ma\
ximum allowed name length is 40", "UID": "df80b906-3937-4668-ac87-e3b3f846e8e3", "allowed": false}`

This package imports `logf "sigs.k8s.io/controller-runtime/pkg/log"`  and create log by this `bundlelog = logf.Log.WithName("bundle-resource")`
So it's possible to change the logger statically.  I'm not sure how to wrap the ValidateCreate function.

Reply:
I believe controller-runtime injects a logger into webhooks if they implement the [`inject.Logger`](https://github.com/kubernetes-sigs/controller-runtime/blob/6627ea2a1c4a7f6741792438e8333e96364028f1/pkg/runtime/inject/inject.go#L151-L155) interface

Reply:
Yes, I looked into it.  It seems that by adding `WithLogger (https://github.com/kubernetes-sigs/controller-runtime/blob/6627ea2a1c4a7f6741792438e8333e96364028f1/pkg/builder/controller.go#L152) in SetupWebhookWithManager, we can override the lodder used by the webhook controller.
What logger should be used in this pacakge?

Reply:
I oversighted.  The WebwookBuilder doesn’t have the function.

Reply:
I would like to handle this in a separate issue.